site/public/js/server.js

```diff
@@ -1625,46 +1625,66 @@ function enableKeyToClick(){
+        confirm_message = `You are flagging ${user_id}'s preferred image as inappropriate.\nThis should be done if the image is not a recognizable passport style photo.\n\nDo you wish to proceed?`;
+        result_message = `${user_id}'s preferred image was successfully flagged.`;
     }
     else {
-        message = `${user_id}'s preferred image has be flagged as inappropriate.\nThis was done because the image is not a recognizable, passport style photo.\n\nYou are reverting to ${user_id}'s preferred image.\nDo you wish to proceed?`;
+        confirm_message = `${user_id}'s preferred image has be flagged as inappropriate.\nThis was done because the image is not a recognizable, passport style photo.\n\nYou are reverting to ${user_id}'s preferred image.\nDo you wish to proceed?`;
+        result_message = `${user_id}'s preferred image was successfully restored.`;
     }
 
-    const confirmed = confirm(message);
+    const confirmed = confirm(confirm_message);
 
     if (confirmed) {
-        const input_elem = document.createElement('input');
-        input_elem.setAttribute('name', 'user_id');
-        input_elem.setAttribute('value', user_id);
-
-        const token_elem = document.createElement('input');
-        token_elem.setAttribute('name', 'csrf_token');
-        token_elem.setAttribute('value', csrfToken);
-
-        const flag_elem = document.createElement('input');
-        flag_elem.setAttribute('name', 'flag');
-        flag_elem.setAttribute('value', flag);
-
-        const form = document.createElement('form');
-        form.setAttribute('method', 'post');
-        form.setAttribute('action', buildCourseUrl(['flag_user_image']));
-        form.appendChild(input_elem);
-        form.appendChild(token_elem);
-        form.appendChild(flag_elem);
-
-        document.body.appendChild(form);
-        form.submit();
+        const url = buildCourseUrl(['flag_user_image']);
+
+        const form_data = new FormData();
+        form_data.append('user_id', user_id);
+        form_data.append('csrf_token', csrfToken);
+        form_data.append('flag', flag);
+
+        const makeRequest = async () => {
+            const response = await fetch(url, {method: 'POST', body: form_data});
+            const result = await response.json();
+
+            if (result.status === 'success') {
+                const data = result.data;
+                const image_container = document.querySelector(`.${user_id}-image-container`);
+
+                // Change image
+                const old_image = image_container.querySelector('img');
+
+                const new_image = document.createElement('img');
+                new_image.setAttribute('alt', data.first_last_username);
+                new_image.setAttribute('src', `data:${data.image_mime_type};base64,${data.image_data}`);
+
+                image_container.removeChild(old_image);
```

Comment:
Say you flag someone's photo and after that, you hit reload. Now you're given with the button to unflag that photo but when you do that you don't see changes in effect right away and need to refresh the page. because this user doesn't have an associated image and this line breaks the code.
But when the user has an image (it's not flagged) you can play with the report button and it shows the changes without reloading.

Reply:
Changed